### PR TITLE
reworked units

### PR DESCRIPTION
**Pull request by saskiad**
Take a look a see what's missing

### CODE REVIEW

**Review by dyf — APPROVED** [review]
Do you want to make a subclasses for angles and positions? e.g.

``` python
class Position3d(AindModel):
    x: float
    y: float
    z: float
    unit: SizeUnit

class Orientation3d(AindModel):
    pitch: float
    yaw: float
    roll: float
    unit: AngleUnit
```

**Review by dyf — CHANGES_REQUESTED** [review]
meant to request changes, for the casing, sorry

File: src/aind_data_schema/procedures.py [hunk 1]
```diff
@@ -254,13 +289,16 @@ class OphysProbe(AindModel):
...
+    core_diameter_unit: str = Field("μm", title="Core diameter unit")
     numerical_aperture: float = Field(..., title="Numerical aperture")
     ferrule_material: Optional[FerruleMaterial] = Field(None, title="Ferrule material")
     targeted_structure: str = Field(..., title="Targeted structure")
     stereotactic_coordinate_ap: float = Field(..., title="Stereotactic coordinate A/P (mm)", units="mm")
     stereotactic_coordinate_ml: float = Field(..., title="Stereotactic coodinate M/L (mm)", units="mm")
     stereotactic_coordinate_dv: float = Field(..., title="Stereotactic coordinate D/V (mm)", units="mm")
+    stereotactic_coordinate_unit: SizeUnit = Field(SizeUnit.MM, title="Sterotactic coordinate unit")
     angle: float = Field(..., title="Angle (deg)", units="deg")
+    angle_unit: AngleUnit = Field(AngleUnit.DEG, title-"Angle unit")
```
Comment (dyf) [inline src/aind_data_schema/procedures.py:301]: typo (`title-`)

File: src/aind_data_schema/device.py [hunk 2]
```diff
@@ -188,11 +223,15 @@ class Filter(Device):
     ]
 
     # optional fields
-    diameter: Optional[float] = Field(None, title="Size (mm)", units="mm")
-    thickness: Optional[float] = Field(None, title="Size (mm)", ge=0)
+    diameter: Optional[float] = Field(None, title="Diameter (mm)", units="mm")
+    diameter_unit: SizeUnit = Field(SizeUnit.MM, title="Diameter unit")
+    thickness: Optional[float] = Field(None, title="Thickness (mm)", ge=0)
+    thickness_unit: SizeUnit = Field(SizeUnit.MM, title-"Thickness unit")
```
Comment (dyf) [inline src/aind_data_schema/device.py:229]: typo (`title-`)

**Review by dyf — CHANGES_REQUESTED** [review]

**Comment by saskiad** [comment]
closes issue #154

**Review by dyf — APPROVED** [review]

**Pull request merged**